### ISSUE TITLE
 Rule S3626: Rule should handle throw in ExpressionBody

### ISSUE THREAD

**Issue by huan086** [SonarSource/sonar-dotnet#3354]
Rule does not trigger for the following

```cs
Func<string> getNonce = () =>
{
    var nonce = httpContextAccessor.HttpContext?.GetNonce();
    if (nonce == null)
    {
        throw new InvalidOperationException("HttpContext is null.");
    }

    return nonce;
};
```

It also does not trigger for the following
```cs
Func<string> getNonce = () =>
{
    return httpContextAccessor.HttpContext?.GetNonce() ?? throw new InvalidOperationException("HttpContext is null.");
};
```

The same code can be written as follows, but Rule S3626 gets triggered

```cs
Func<string> getNonce = () => httpContextAccessor.HttpContext?.GetNonce() ?? throw new InvalidOperationException("HttpContext is null.");
```

`warning S3626: Remove this redundant jump.`

**Comment by pavel-mikula-sonarsource** [comment]
Hi @huan086,

Thank you for reporting this issue. We can confirm it as false positive. We already have it reported in #3105 so I'll close this one.
